src/main/kotlin/com/bardsoftware/backend/fleet/rmarkdown/Subscriber.kt

```diff
@@ -41,16 +59,56 @@ internal class MessageReceiverExample(private val callback: (message: String, md
     }
 
     fun processMessage(message: PubsubMessage) {
-        val messageContent = message.data.toStringUtf8()
-        val md5sum = this.dockerProcessor.getMd5Sum(messageContent)
+        val request = CompilerFleet.CompilerFleetRequest.parseFrom(message.data)
+
+        val destination = this.tasksDir.resolve(request.taskId).resolve("files")
+        val zipStream = ZipInputStream(ByteArrayInputStream(request.zipBytes.toByteArray()))
+        var entry: ZipEntry? = zipStream.nextEntry
+
+        while (entry != null) {
+            val filename = entry.name
+            val newFile = destination.resolve(filename).toFile()
+
+            if (!newFile.parentFile.mkdirs()) {
+                throw IOException("unable to create dirs while unzipping")
```

Comment:
When you look at the logs, you almast always want to see something more informative than "ooops, somethign went wrong". What is the id of the task which caused this exception? Directory name?

Reply:
Yes, it's a good remark.
Done